api/utils/overlay.py

```diff
@@ -1,30 +1,72 @@
+import asyncio
+
 import pandas as pd
 import geopandas as gpd
-import os
 
+from api.config import PROJECT_ROOT, PADDING_CONFIG, MASTER_DATA_FN, LAYERS
+from api.config import Status
+
+from concurrent.futures import ProcessPoolExecutor
+
+def overlay_layer(
+    padding_config: dict[dict[str, float]],
+    master_data: gpd.GeoDataFrame,
+    category_name: str,
+    layer: str,
+):
+    file_path = PROJECT_ROOT / 'data' / 'processed' / f'{layer}.geojson'
+    data = gpd.read_file(file_path)
+    buffer_df = apply_buffer_from_config(padding_config=padding_config)
```

Comment:
tu treba pridat `data = data[~data['geometry'].isnull()]`

Reply:
medzi riadky 18 a 19

Reply:
to sa od vtedy nejak posunulo, musis mi to este raz komentnut na tej lajne kde to chces